### PR TITLE
[Merged by Bors] - Make openssl-sys always vendored and add windows support

### LINKED ISSUE #79: Windows support for OpenSSL
Usage of `RawFd` prevents compilation on windows.

We should add windows compilation in CI to catch this.

### PR DESCRIPTION
**Pull request by simlay**
Closes #79.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -31,12 +31,14 @@ pub mod net;
...
+
 #[cfg(all(unix, feature = "rust_tls", not(feature = "native2_tls")))]
 pub use rust_tls as tls;
 
-#[cfg(all(unix, feature = "native2_tls"))]
+#[cfg(all(any(unix, windows), feature = "native2_tls"))]
 pub mod native_tls;
-#[cfg(all(unix, feature = "native2_tls", not(feature = "rust_tls")))]
+
+#[cfg(all(any(unix, windows), feature = "native2_tls", not(feature = "rust_tls")))]
```
Comment (simlay) [inline src/lib.rs:41]: This was needed to get k8-client working over in https://github.com/infinyon/fluvio/pull/1218 but this breaks CI here :/

File: Cargo.toml [hunk 2]
```diff
@@ -78,7 +78,7 @@ fluvio_ws_stream_wasm = "0.7.0"
 bytes = "1.0.0"
 lazy_static = "1.2.0"
 num_cpus = "1.10.1"
-futures-util = { version = "0.3.5", features = ["sink"] }
+futures-util = { version = "0.3.5", features = ["sink", "io"] }
```
Comment (sehz) [inline Cargo.toml:81]: do we need io everywhere or can be scoped?  probably same for `sink` as well

**Review by sehz — CHANGES_REQUESTED** [review]
Very nice.  a minor nit.  also can you see if you can update changelog as well?

File: Cargo.toml (hunk 2 above)
Comment (simlay) [reply Cargo.toml:87]: Huh. The io flag was the one that was actually needed. I made it only used for the tls features.

**Comment by simlay** [comment]
> Very nice. a minor nit. also can you see if you can update changelog as well?

Done.

**Comment by sehz** [comment]
approved

**Pull request closed without merging**